python/lsst/ts/genericcamera/driver/alliedvisioncamera.py

```diff
@@ -0,0 +1,302 @@
+                self.camera.OffsetY.set(top)
+                self.camera.Width.set(width)
+                self.camera.Height.set(height)
+
+    def set_full_frame(self):
+        """Sets the region of interest to the whole sensor."""
+        with vimba.Vimba.get_instance():
+            with self.camera:
+                self.set_roi(
+                    0, 0, self.camera.WidthMax.get(), self.camera.HeightMax.get()
+                )
+
+    def start_live_view(self):
+        """Configure the camera for live view.
+
+        This should change the image format to 8bits per pixel so
+        the image can be encoded to JPEG."""
+
+        self.is_live_exposure = True
+        with vimba.Vimba.get_instance():
+            with self.camera:
+                self.camera.AcquisitionMode.set("SingleFrame")
+                self.camera.ExposureAuto.set("Continuous")
+                self.camera.set_pixel_format(vimba.PixelFormat.Mono8)
+        super().start_live_view()
+
+    def stop_live_view(self):
+        """Stops an active live view data stream from the camera.
+
+        This should review the image format back to original format."""
+        self.is_live_exposure = False
+        with vimba.Vimba.get_instance():
+            with self.camera:
+                self.camera.set_pixel_format(self.normal_image_type)
+        super().stop_live_view()
+
+    def _set_camera_exposure_time(self):
+        """Set the exposure time on the camera."""
+        with vimba.Vimba.get_instance():
+            with self.camera:
+                if not self.is_live_exposure or not self.liveview_use_autoexposure:
+                    self.log.debug("Setting camera exposure time")
+                    self.camera.ExposureAuto.set("Off")
+                    self.camera.ExposureTimeAbs.set(self.exposure_time * MICROSECONDS)
+
+    async def start_take_image(self, exp_time, shutter, science, guide, wfs):
+        """Start taking an image or a set of images.
+
+        Parameters
+        ----------
+        exp_time : float
+            The exposure time in seconds.
+        shutter : bool
+            Should the shutter be opened?
+        science : bool
+            Should the science/main sensor be used?
+        guide : bool
+            Should guider sensor be used?
+        wfs : bool
+            Should wave front sensor be used?
+        """
+        self.exposure_time = exp_time
+        await self.loop.run_in_executor(self.executor, self._set_camera_exposure_time)
+        await super().start_take_image(exp_time, shutter, science, guide, wfs)
+
+    def _get_frame(self):
+        """Wait for the frame to arrive from the camera.
+
+        Returns
+        -------
+        Vimba.Frame
+            The frame captured from the camera.
+        """
+        with vimba.Vimba.get_instance():
+            # v.enable_log(vimba.LOG_CONFIG_INFO)
+            with self.camera:
+                if self.liveview_use_autoexposure:
+                    timeout = int(
```

Comment:
Ok, I see that your `MILLISECONDS` is actually supposed to be `1/ms`.. I would recommend changing the definition because it is a bit confusing. 

Also, consider renaming this variable `timeout_ms`

Reply:
Changed the local variable name and hopefully the new constant names make more sense.